src/server/middleware/nydus/nydus.go

```diff
@@ -0,0 +1,158 @@
+)
+
+var (
+	mediaTypeNydusLayer = "application/vnd.oci.image.layer.nydus.blob.v1"
+)
+
+// NydusAcceleratorMiddleware middleware to record the linkeage of artifact and its accessory
+/*
+/v2/library/hello-world/manifests/latest-nydus
+{
+    "schemaVersion": 2,
+    "config": {
+      "mediaType": "application/vnd.oci.image.config.v1+json",
+      "digest": "sha256:f7d0778a3c468a5203e95a9efd4d67ecef0d2a04866bb3320f0d5d637812aaee",
+      "size": 466
+    },
+    "layers": [
+      {
+        "mediaType": "application/vnd.oci.image.layer.nydus.blob.v1",
+        "digest": "sha256:fd9923a8e2bdc53747dbba3311be876a1deff4658785830e6030c5a8287acf74 ",
+        "size": 3011,
+        "annotations": {
+          "containerd.io/snapshot/nydus-blob": "true"
+        }
+      },
+      {
+        "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+        "digest": "sha256:d49bf6d7db9dac935b99d4c2c846b0d280f550aae62012f888d5a6e3ca59a589",
+        "size": 459,
+        "annotations": {
+            "containerd.io/snapshot/nydus-blob-ids": "[\"fd9923a8e2bdc53747dbba3311be876a1deff4658785830e6030c5a8287acf74\"]",
+            "containerd.io/snapshot/nydus-bootstrap": "true",
+            "containerd.io/snapshot/nydus-rafs-version": "5"
+        }
+      }
+    ],
+    "annotations": {
+        "io.goharbor.artifact.v1alpha1.acceleration.driver.name":"nydus",
+        "io.goharbor.artifact.v1alpha1.acceleration.driver.version":"5",
+        "io.goharbor.artifact.v1alpha1.acceleration.source.digest":"sha256:f54a58bc1aac5ea1a25d796ae155dc228b3f0e11d046ae276b39c4bf2f13d8c4"
+    }
+}
+
+*/
+func AcceleratorMiddleware() func(http.Handler) http.Handler {
+	return middleware.AfterResponse(func(w http.ResponseWriter, r *http.Request, statusCode int) error {
+		if statusCode != http.StatusCreated {
+			return nil
+		}
+
+		log.Info("Start NydusAccelerator Middleware")
+		ctx := r.Context()
+		logger := log.G(ctx).WithFields(log.Fields{"middleware": "nydus"})
+
+		none := lib.ArtifactInfo{}
+		info := lib.GetArtifactInfo(ctx)
+		if info == none {
+			return errors.New("artifactinfo middleware required before this middleware").WithCode(errors.NotFoundCode)
+		}
+		if info.Tag == "" {
+			return nil
+		}
+		fmt.Println("info: ", info)
+		fmt.Println("r.URL.Path:", r.URL.Path)
+
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return err
+		}
+
+		contentType := r.Header.Get("Content-Type")
+		manifest, desc, err := distribution.UnmarshalManifest(contentType, body)
+		fmt.Println("manifest:", manifest)
+		fmt.Println("descriptor:", desc)
+
+		if err != nil {
+			logger.Errorf("unmarshal manifest failed, error: %v", err)
+			return err
+		}
+		fmt.Println("manifest References", manifest.References())
+
+		var isNydus bool
+		for _, descriptor := range manifest.References() {
+			log.Info("descriptor.Annotations", descriptor.Annotations)
+			log.Info("mediaTypeNydusLayer", mediaTypeNydusLayer)
+			annotationMap := descriptor.Annotations
```

Comment:
IMO, no need to create this new variable as only read value from `descriptor.Annotations`.